### PR TITLE
Add CurlHandler.ConnectTimeout

### PR DESCRIPTION
**Pull request by stephentoub**
Adds a timeout to the connection phase of CurlHandler.

cc: @davidsh, @cipop, @ericeil, @kapilash, @bartonjs, @joshfree

### CODE REVIEW

File: src/Common/src/System/Net/Http/HttpHandlerDefaults.cs [hunk 1]
```diff
@@ -16,5 +16,6 @@ internal static class HttpHandlerDefaults
         public const bool DefaultPreAuthenticate = false;
         public const ClientCertificateOption DefaultClientCertificateOption = ClientCertificateOption.Manual;
         public const bool DefaultUseProxy = true;
+        public readonly static TimeSpan DefaultConnectTimeout = TimeSpan.FromSeconds(60);
```
Comment (justinvp) [inline src/Common/src/System/Net/Http/HttpHandlerDefaults.cs]: Nit: Which is preferred: `readonly static` or `static readonly`?

File: src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs [hunk 2]
```diff
@@ -58,6 +58,9 @@ internal partial class CurlHandler : HttpMessageHandler
             new KeyValuePair<string,CURLAUTH>("Basic", CURLAUTH.Basic),
         };
 
+        // Max timeout value used by WinHttp handler, so mapping to that here.
+        private static readonly TimeSpan s_maxTimeout = TimeSpan.FromMilliseconds(int.MaxValue);
```
Comment (justinvp) [inline src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs]: Nit: Which is preferred: `readonly static` or `static readonly`? Should it be consistent with the rest of the static readonly fields below?

File: src/Common/src/System/Net/Http/HttpHandlerDefaults.cs (hunk 1 above)
Comment (ericeil) [inline src/Common/src/System/Net/Http/HttpHandlerDefaults.cs]: This looks like it'll add a .cctor to this type, which didn't have one before.  I wonder if it's worth it.  What about making it a property instead:

```csharp
public static TimeSpan DefaultConnectTimeout => TimeSpan.FromSeconds(60);
```

**Comment by ericeil** [comment]
Aside from the one minor comment, LGTM.

File: src/Common/src/System/Net/Http/HttpHandlerDefaults.cs (hunk 1 above)
Comment (stephentoub) [inline src/Common/src/System/Net/Http/HttpHandlerDefaults.cs]: Makes sense.  I'll change it.

Comment (stephentoub) [inline src/Common/src/System/Net/Http/HttpHandlerDefaults.cs]: > Which is preferred: readonly static or static readonly?

There are 164 occurrences of ```readonly static``` across the src tree, and 2343 occurrences of ```static readonly```, so looks like the latter.  I'll fix this to match.

File: src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs]: ```static readonly``` is preferred.

File: src/Common/src/System/Net/Http/HttpHandlerDefaults.cs (hunk 1 above)
Comment (davidsh) [inline src/Common/src/System/Net/Http/HttpHandlerDefaults.cs]: >There are 164 occurrences of readonly static across the src tree, and 2343 occurrences of static readonly, so looks like the latter. I'll fix this to match.

We discussed this months ago and @weshaggard updated the coding-guidelines with the answer to this.

Comment (stephentoub) [inline src/Common/src/System/Net/Http/HttpHandlerDefaults.cs]: > We discussed this months ago and @weshaggard updated the coding-guidelines with the answer to this.

I did, actually :)
https://github.com/dotnet/corefx/commit/88c467b94aeed05c07fa14cc2769f49d66125ab3
based on the same reasoning.  I just didn't follow it here due to copy-pasting from existing code and not noticing it was the wrong order.

**Pull request merged**